runtime/minor_gc.c

```diff
@@ -413,7 +413,7 @@ static void oldify_mopup (struct oldify_state* st, int do_ephemerons)
          re < ephe_ref_table.ptr; re++) {
       value *data = re->offset == CAML_EPHE_DATA_OFFSET
               ? &Ephe_data(re->ephe)
-              :  &Field(re->ephe, re->offset);
+              : (value*)&Field(re->ephe, re->offset);
```

Comment:
Here you could avoid the cast by declaring `volatile value * data`.  There would be an implicit conversion from non-volatile to volatile in the `Ephe_data` branch.

Reply:
This change necessitated changing the type of the `value*` argument in `oldify_one` to `volatile value*`.

Reply:
Looks like a reasonable change to me.